src/server/http/xsrf.js

```diff
@@ -4,18 +4,49 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
 
   server.ext('onPostAuth', function (req, reply) {
     if (disabled) {
       return reply.continue();
     }
 
     const isSafeMethod = req.method === 'get' || req.method === 'head';
-    const hasVersionHeader = versionHeader in req.headers;
-    const hasXsrfHeader = xsrfHeader in req.headers;
+    if (!isSafeMethod) {
```

Comment:
Ah, right, forgot about the disabled check :/ Sure, will add an early return then. Once we get rid of deprecation related stuff it will look nicer anyway.